types/readdir-glob/index.d.ts

```diff
@@ -0,0 +1,129 @@
+// Type definitions for readdir-glob 1.1
+// Project: https://github.com/Yqnn/node-readdir-glob
+// Definitions by: Dolan Miu <https://github.com/dolanmiu>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as fs from 'fs';
+import { EventEmitter } from 'events';
+
+declare function readdirGlob(root: string, options: readdirGlob.Options): readdirGlob.ReaddirGlob;
+
+declare namespace readdirGlob {
+    interface Options {
+        /**
+         * Glob pattern or Array of Glob patterns to match the found files with. A file has to match at least one of the provided patterns to be returned.
+         */
+        pattern?: string;
+        /**
+         * Allow pattern to match filenames starting with a period, even if the pattern does not explicitly have a period in that spot.
+         */
+        dot?: boolean;
+        /**
+         * Disable `**` matching against multiple folder names.
+         */
+        noglobstar?: boolean;
+        /**
+         * Perform a basename-only match if the pattern does not contain any slash characters. That is, `*.js` would be treated as equivalent to `**\/*.js`, matching all js files in all directories.
+         */
+        matchBase?: boolean;
+        /**
+         * Perform a case-insensitive match. Note: on case-insensitive file systems, non-magic patterns will match by default, since `stat` and `readdir` will not raise errors.
+         */
+        nocase?: boolean;
+        /**
+         * Glob pattern or Array of Glob patterns to exclude matches. If a file or a folder matches at least one of the provided patterns, it's not returned.
+         * It doesn't prevent files from folder content to be returned. Note: ignore patterns are always in dot:true mode.
+         */
+        ignore?: boolean;
```

Comment:
This change (at least) breaks my archiver integration via [this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64611) since `ignore` is supposed to be `string | string[] | IgnoreLike`. See https://github.com/isaacs/node-glob/blob/main/src/glob.ts#L118

Reply:
From looking at the source of readdir-glob, I can see that it is `string | string[]`:

https://github.com/Yqnn/node-readdir-glob/blob/master/index.js#L145-L147

Minimatch takes a string as its constructor arg:

https://github.com/isaacs/minimatch/blob/main/src/index.ts#L180